openpype/hosts/fusion/plugins/publish/validate_frame_range.py

```diff
@@ -0,0 +1,79 @@
+import pyblish.api
+
+from openpype.pipeline import (
+    PublishValidationError,
+    OptionalPyblishPluginMixin,
+)
+from openpype.pipeline.publish import RepairContextAction
+
+from openpype.pipeline.context_tools import get_current_project_asset
+from openpype.hosts.fusion.api.lib import (
+    set_asset_framerange,
+)
+
+
+class ValidateFrameRange(
+    pyblish.api.ContextPlugin,
+    OptionalPyblishPluginMixin,
+):
+    """Validate if the comp has the correct frame range"""
+
+    order = pyblish.api.ValidatorOrder
+    label = "Validate Frame Range"
+    families = ["render"]
+    hosts = ["fusion"]
+    optional = True
+    actions = [RepairContextAction]
+
+    def process(self, context):
+        if not self.is_active(context.data):
+            return
+
+        asset_doc = get_current_project_asset()
+        start = asset_doc["data"]["frameStart"]
+        end = asset_doc["data"]["frameEnd"]
+        handle_start = asset_doc["data"]["handleStart"]
+        handle_end = asset_doc["data"]["handleEnd"]
+
+        # Convert any potential none type to zero
+        handle_start = handle_start or 0
+        handle_end = handle_end or 0
+
+        # Calcualte in/out points
+        range_start = start - handle_start
+        range_end = end - handle_start
+
+        invalid = self.get_invalid(context, range_start, range_end)
+        if invalid:
+            raise PublishValidationError(
+                "The comp's frame range isn't correct"
+                "compared to the projects properties."
```

Comment:
```suggestion
                "compared to the asset's properties."
```

Also should this report what the difference is? What are the current values and what are the asset's values?

Reply:
Fixed.
It also do report the difference. You see "Inpoint is set to {} but should start at {}" at line 61 for example.
The way it's setup up you only get the needed info if only one of the in/out points are wrong.